### PR TITLE
Minor improvements to the Stripe subscription cancellation flow

### PR DESCRIPTION
**Pull request by jankeromnes**
## Description

These are the definitely-valuable parts salvaged from my [proration effort](https://github.com/gitpod-io/gitpod/pull/14148) (which ended up not working out):

- Immediately issue a final invoice when cancelling Stripe subscriptions from Gitpod (currently, we're letting Stripe wait until the end of the month)

- When deleting a user account, also cancel any individual Stripe subscription (we forgot to do this until now)

## Related Issue(s)

Fixes #

## How to test

1. Create a team called "Gitpod [Something]"
2. Go to `/billing`
3. Upgrade your user to usage-based billing by adding a payment method
4. Then, delete your user -- this should both:
  a. automatically cancel your individual Stripe subscription (double-check in the Stripe dashboard)
  b. immediately issue a final invoice (it's going to be in draft for 1h, but at least Stripe won't wait until the end of the month to create it)

## Release Notes

```release-note
NONE
```

## Werft options:

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`

### CODE REVIEW

**Comment by laushinka** [comment]
I deleted my account and the subscription still says `Active` 15 minutes later. Does it take a while?

**Comment by jankeromnes** [comment]
Many thanks @laushinka for testing! Subscription cancellation on user deletion should happen immediately.

Meanwhile, the preview environment died, so I'll rebase this PR so that we can test it again.

File: components/server/ee/src/user/stripe-service.ts [hunk 1]
```diff
@@ -117,7 +117,7 @@ export class StripeService {
 
     async cancelSubscription(subscriptionId: string): Promise<void> {
         await reportStripeOutcome("subscriptions_cancel", () => {
-            return this.getStripe().subscriptions.del(subscriptionId);
+            return this.getStripe().subscriptions.del(subscriptionId, { invoice_now: true });
```
Comment (easyCZ) [inline components/server/ee/src/user/stripe-service.ts:120]: Do we also need to bring this change into the `stripeService.cancelSubscription()` on the usage component?

**Review by easyCZ — COMMENTED** [review]
LGTM, one Q

File: components/server/ee/src/user/stripe-service.ts (hunk 1 above)
Comment (easyCZ) [reply components/server/ee/src/user/stripe-service.ts:120]: Do you wanna add it in this PR or a follow-up?

**Comment by laushinka** [comment]
> Could you please also try again when you have some time? 😊 🙏

Turns out my mistake last time was subscribing the team and not my individual account 🙈
Works well! Approving and holding for the question above[[1](https://github.com/gitpod-io/gitpod/pull/14265#pullrequestreview-1172097937)].

File: components/server/ee/src/user/stripe-service.ts (hunk 1 above)
Comment (easyCZ) [reply components/server/ee/src/user/stripe-service.ts:120]: SGTM, unhold at will

Comment (jankeromnes) [reply components/server/ee/src/user/stripe-service.ts:120]: @easyCZ Would you have a direct link to the code location you meant by "the `stripeService.cancelSubscription()` on the usage component?"

I can't find the relevant code, and in fact we never seem to actually cancel Stripe subscriptions from the usage component? 🤔 (I only found [this function called CancelSubscription](https://github.com/gitpod-io/gitpod/blob/ee8354aa79e550f836e6d4f461d85fb59134e32b/components/usage/pkg/apiv1/billing.go#L337-L365), but it does not seem to call Stripe and only updates our internal models like the CostCenter).

Comment (jankeromnes) [reply components/server/ee/src/user/stripe-service.ts:120]: Also, when grepping the code, the only Go-side Stripe API methods we call are `Subscriptions.New` (1 occurrence) and `Subscriptions.Get` (1 occurrence). No `Subscriptions.Cancel` anywhere.

**Pull request merged**